runtime/compiler/control/JITaaSCompilationThread.cpp

```diff
@@ -2603,19 +2587,27 @@ remoteCompile(
       resolvedMirrorMethodsPersistIPInfo = std::get<7>(recv);
       if (statusCode >= compilationMaxError)
          throw JITaaS::StreamTypeMismatch("Did not receive a valid TR_CompilationErrorCode as the final message on the stream.");
+      if (statusCode != compilationStreamVersionIncompatible)
+         JITaaS::J9ClientStream::setVersionCheckStatus(JITaaS::COMPATIBLE);
+
       status = client->waitForFinish();
       }
    catch (const JITaaS::StreamFailure &e)
       {
-      if (TR::Options::getVerboseOption(TR_VerboseJITaaS))
-         TR_VerboseLog::writeLineLocked(TR_Vlog_JITaaS, e.what());
-         
       client->~J9ClientStream();
       TR_Memory::jitPersistentFree(client);
       compInfoPT->setClientStream(NULL);
 
       compiler->failCompilation<JITaaS::StreamFailure>(e.what());
       }
+   catch (const JITaaS::StreamVersionIncompatible &e)
+      {
```

Comment:
Disconnecting the incompatible stream is the right thing to do. We just have to make sure the client does not try over and over to connect to an incompatible server.

Reply:
the reason i used a static status for compatibility is to make sure the client does not try to connect over and over again, I failed to see how that can be achieved with a non static compatibility status.